lib/src/lifecycle_module.dart

```diff
@@ -19,6 +19,7 @@ import 'dart:async';
 import 'package:logging/logging.dart';
 import 'package:meta/meta.dart' show protected, required;
 import 'package:w_common/disposable.dart';
+import 'package:w_module/w_module.dart';
```

Comment:
For files that are in the `src/` directory, I usually try to avoid imports like this where you import your own entry point that contains everything (including this file) instead of just the piece that you need (`SimpleModule`).

You could instead import:
```dart
import 'package:w_module/src/simple_module.dart';
```